src/Composer/Plugin.php

```diff
@@ -85,72 +91,50 @@ public static function getSubscribedEvents() {
     return array(
       PackageEvents::POST_PACKAGE_INSTALL => "onPostPackageEvent",
       PackageEvents::POST_PACKAGE_UPDATE => "onPostPackageEvent",
-      ScriptEvents::PRE_INSTALL_CMD => array(
-        array('scaffoldComposerIncludes', self::CALLBACK_PRIORITY),
-        array('checkInstallerPaths'),
-      ),
       ScriptEvents::POST_UPDATE_CMD => array(
-        array('scaffoldComposerIncludes', self::CALLBACK_PRIORITY),
         array('onPostCmdEvent'),
       ),
-      ScriptEvents::PRE_AUTOLOAD_DUMP => array(
-        array('scaffoldComposerIncludes', self::CALLBACK_PRIORITY),
-      ),
     );
   }
 
   /**
-   * Verify that composer.json contains correct values for installer-paths.
-   *
-   * Unfortunately, these values cannot be placed in composer.required.json.
-   *
-   * @see https://github.com/wikimedia/composer-merge-plugin/issues/139
    *
-   * @param \Composer\Script\Event $event
    */
-  public function checkInstallerPaths(Event $event) {
-    $extra = $this->composer->getPackage()->getExtra();
-    if (empty($extra['installer-paths'])) {
-      $this->io->write('<error>Error: extra.installer-paths is missing from your composer.json file.</error>');
-    }
-    else {
-      $composer_required_json_filename = $this->getVendorPath() . '/acquia/blt/template/composer.json';
-      if (file_exists($composer_required_json_filename)) {
-        $composer_required_json = json_decode(file_get_contents($composer_required_json_filename), TRUE);
-        if ($composer_required_json['extra']['installer-paths'] != $extra['installer-paths']) {
-          $this->io->write('<warning>Warning: The value for extra.installer-paths in composer.json differs from BLT\'s recommended values.</warning>');
-          $this->io->write('<warning>See ' . $composer_required_json_filename . '</warning>');
-        }
+  public function importDefaultConfig() {
+    $extra = $this->rootPackage->getExtra();
+    $composer_template_filename = $this->getVendorPath() . '/acquia/blt/template/composer.json';
+    if (file_exists($composer_template_filename)) {
+      $composer_template = json_decode(file_get_contents($composer_template_filename), TRUE);
+
+      // 'extra' array.
+      $merged_extra = ArrayManipulator::arrayMergeRecursiveDistinct($composer_template['extra'], $extra);
```

Comment:
We're not actually testing this because our tests just use the template. Need to resolve that.